### PR TITLE
feat: Allow single Testkube instance to watch for Test,s TestSources and TestTriggers across all namespaces

### PR DESCRIPTION
**Pull request by cedricmckinnie**
## Pull request description 

Original thread to avoid duplicating comments:
https://github.com/kubeshop/testkube/issues/3804#issuecomment-1613552389

## Checklist (choose whats happened)

- [ ] breaking change! (describe)
- [ ] tested locally
- [ ] tested on cluster
- [ ] added new dependencies
- [ ] updated the docs
- [x] added a test

## Breaking changes

-

## Changes

- Allow single Testkube instance to watch for Test,s TestSources and TestTriggers across all namespaces

## Fixes

-

### CODE REVIEW

**Comment by cedricmckinnie** [comment]
What would be the simplest way to test locally/in-cluster from my dev setup? Are there any docs I can follow?

**Review by vsukhin — APPROVED** [review]
lgtm, @dejanzele ?

File: pkg/triggers/watcher.go [hunk 1]
```diff
@@ -41,13 +41,12 @@ type k8sInformers struct {
...
-	testSuiteInformer   testkubeinformerv2.TestSuiteInformer
-	testInformer        testkubeinformerv3.TestInformer
+	testTriggerInformers []testkubeinformerv1.TestTriggerInformer
+	testSuiteInformers   []testkubeinformerv2.TestSuiteInformer
+	testInformers        []testkubeinformerv3.TestInformer
 }
 
-func newK8sInformers(clientset kubernetes.Interface, testKubeClientset versioned.Interface,
-	testkubeNamespace string, watcherNamespaces []string) *k8sInformers {
+func newK8sInformers(clientset kubernetes.Interface, testKubeClientset versioned.Interface, testkubeNamespace string, watcherNamespaces []string, watchTestkubeCrAllNamespaces bool) *k8sInformers {
```
Comment (vsukhin) [inline pkg/triggers/watcher.go:49]: where do you use watchTestkubeCrAllNamespaces?

Comment (cedricmckinnie) [reply pkg/triggers/watcher.go:49]: My mistake. I meant to make it optional as some users might not want this functionality.

https://github.com/kubeshop/testkube/pull/4106/files#diff-59b1352b84890f4bab146e9c47bb21322ac161fd777114091c762d61838ca3efR68-R73

File: pkg/triggers/watcher.go [hunk 2]
```diff
@@ -63,13 +62,21 @@ func newK8sInformers(clientset kubernetes.Interface, testKubeClientset versioned
...
+	var watchedNamespaces string
+	if watchTestkubeCrAllNamespaces == true {
+		watchedNamespaces = v1.NamespaceAll
+	} else {
+		watchedNamespaces = testkubeNamespace
+	}
+	var testkubeInformerFactory externalversions.SharedInformerFactory
+	testkubeInformerFactory = externalversions.NewSharedInformerFactoryWithOptions(
+		testKubeClientset, 0, externalversions.WithNamespace(watchedNamespaces))
+	k8sInformers.testTriggerInformers = append(k8sInformers.testTriggerInformers, testkubeInformerFactory.Tests().V1().TestTriggers())
```
Comment (vsukhin) [inline pkg/triggers/watcher.go:77]: do we need an array of informers in this case?

**Comment by cedricmckinnie** [comment]
Canceling. Will allow your team to decide at the business level if this is something needed. Will open from another branch as well if needed

**Pull request closed without merging**